### PR TITLE
paste links into richtext

### PR DESCRIPTION
**Pull request by beliayeu**

### CODE REVIEW

File: packages/bodiless-richtext/src/serializers/htmlSerializer.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+const deserializeHtml = (
+  html: string,
+  deserializers: Deserializer[],
+) => {
+  const parsed = new DOMParser().parseFromString(html, 'text/html');
+  return deserializeElement(parsed.body, deserializers);
+};
+
```
Comment (beliayeu) [inline packages/bodiless-richtext/src/serializers/htmlSerializer.ts:98]:
```
export const createDeserializer(nodeName: string, nodeMapper: (element: HTMLElement) => SlateNode) => ({
  htmlElementMatcher: (element: HTMLElement) => element.nodeName === nodeName,
  htmlElementToNodeMapper: nodeMapper,
});
```

File: packages/bodiless-richtext/src/serializers/htmlSerializer.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const createLinkDeserializer = () => ({
+  htmlElementMatcher: (element: HTMLElement) => element.nodeName === 'A',
```
Comment (beliayeu) [inline packages/bodiless-richtext/src/serializers/htmlSerializer.ts]:
```
const createLinkDeserializer = createDeserializer(
 'A',
 (element: HTMLElement) => ({
    type: 'Link',
    data: { slatenode: { href: element.getAttribute('href') } },
  }),
);
```

File: packages/bodiless-richtext/src/serializers/htmlSerializer.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  htmlElementToNodeMapper: (element: HTMLElement) => ({
+    type: 'Link',
+    data: { slatenode: { href: element.getAttribute('href') } },
+  }),
+});
+
```
Comment (beliayeu) [inline packages/bodiless-richtext/src/serializers/htmlSerializer.ts:103]: export const createDeserializers = (deserializerNames?: string[]) => {
  ... 
 conditionally do createLinkDeserializer
  ...
}

File: examples/test-site/src/scripts/sample-html-richtext-serialization.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+const inputHtml = `
+ <div>
+   <p>some test text</p>
+   <a href="/testhref">TestHref</a>
+   <p>additional text</p>
+ </div>
+`;
+
+const deserializers = [
```
Comment (beliayeu) [inline examples/test-site/src/scripts/sample-html-richtext-serialization.js]:
```
import { createDefaultDeserializer, createDeserializer } from '@bodiless/richtext';

const deserializers = [
 ...createDefaultDeserializer,
  // if I want to override
  Link: createDeserializer(
    'A', 
 (element: HTMLElement) => ({
    type: 'Link',
    data: { slatenode: { href: element.getAttribute('href'), 'aria-label': element.getAttribute('aria-label') } },
  }),
]
```

File: examples/test-site/src/scripts/sample-html-richtext-serialization.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+const {
+  deserializeHtml,
+  createLinkDeserializer,
+  createHeader2Deserializer,
+} = require('@bodiless/richtext/lib/serializers');
+
+const inputHtml = `
+ <div>
+   <p>some test text</p>
```
Comment (beliayeu) [inline examples/test-site/src/scripts/sample-html-richtext-serialization.js]:
```
<p>test paragraph</p> <-- Richtext ->
<ul> <-- List component -->
 <li><a href='/test/'>Test</a></li>
 <li><a href='/test2/'>Test2</a></li>
</ul>
<p>another test paragraph</p> <-- Richtext -->
```

ul -> paragraph
li -> paragraph
a -> Link

File: packages/bodiless-richtext/src/RichText.tsx [hunk 6]
```diff
@@ -231,6 +233,7 @@ const BasicRichText = React.memo(<P extends object>(props: P & RichTextProps) =>
       finalComponents: finalComponents$,
       plugins: getPlugins(finalComponents$),
       globalButtons: getGlobalButtons(finalComponents$),
+      deserializers: getDeserializers(finalComponents$),
```
Comment (dewen) [inline packages/bodiless-richtext/src/RichText.tsx]: do we need this deserializers here?

Comment (beliayeu) [reply packages/bodiless-richtext/src/RichText.tsx]: nope, it is not needed here
we can remove it

**Comment by beliayeu** [comment]
pr moved to the main repo https://github.com/johnsonandjohnson/Bodiless-JS/pull/769

**Pull request closed without merging**